tic-tac-toe-ts/methods.ts

```diff
@@ -0,0 +1,297 @@
+import * as readlineSync from "readline-sync";
+import {
+  USER,
+  COMPUTER,
+  USER_DEFAULT_ICON,
+  COMP_DEFAULT_ICON,
+} from "./constants";
+
+export function getValidBoardSize(): number {
+  let attempts: number = 1;
+  let userInput: number = parseInt(
+    readlineSync.question("what will be the size of matrix : ")
+  );
+  while (!isValidBoardSize(userInput) && attempts < 3) {
+    userInput = parseInt(
+      readlineSync.question(
+        `Not a valid matrix size for this game, please enter number greater than 2. You have ${
+          3 - attempts
+        }/3 attempts left, else we'll choose 3 : `
+      )
+    );
+    attempts += 1;
+  }
+
+  userInput = attempts > 2 ? 3 : userInput;
+
+  console.log(`You have selected the matrix: ${userInput} x ${userInput} `);
+
+  return userInput;
+}
+
+function isValidBoardSize(size: number): boolean {
+  if (isNaN(size) || size < 3) {
+    return false;
+  }
+  return true;
+}
+
+export function createBoard(
+  size: number
+): [number[][], { [key: number]: number[] }, number[]] {
```

Comment:
- in case of complex types, create a type elsewhere 9maybe a different file or at the top of this file) and then use it here
- that way, you can give it proper names too
- example below:

```typescript
type CreateBoardResponse = {
  matrix: number[][];
  valueIndicesMatch: { [key: number]: number[] };
  availableCellIndices: number[]
}
```